### PR TITLE
Canvas.drawShadow API based on SkShadowUtils::drawShadow

### PR DESCRIPTION
**Pull request by jason-simmons**

### CODE REVIEW

**Comment by jason-simmons** [comment]
@abarth

File: lib/ui/painting.dart [hunk 1]
```diff
@@ -1548,6 +1554,33 @@ class Canvas extends NativeFieldWrapperClass2 {
                   Int32List colors,
                   int blendMode,
                   Float32List cullRect) native "Canvas_drawAtlas";
+
+  void drawShadow(Path path, Color color, {
+    double occluderHeight: 0.0,
+    double lightPosX: 0.0,
+    double lightPosY: 0.0,
+    double lightPosZ: 0.0,
```
Comment (abarth) [inline lib/ui/painting.dart]: s/Pos/Position/  (please use complete words in identifiers)

File: lib/ui/painting.dart [hunk 2]
```diff
@@ -1548,6 +1554,33 @@ class Canvas extends NativeFieldWrapperClass2 {
                   Int32List colors,
                   int blendMode,
                   Float32List cullRect) native "Canvas_drawAtlas";
+
+  void drawShadow(Path path, Color color, {
```
Comment (abarth) [inline lib/ui/painting.dart]: Needs docs.  Don't forget to cross-link with ShadowFlags.

I wonder if we should simplify this API to match the level of control we give with physical model layer.  The client code is going to toggle between the two and want the same output.  If we give this low level interface, the client will need to know the details of our physical model implementation to match the parameters.  If we use the same interface, then we can side-step that a bit.

**Comment by jason-simmons** [comment]
Reduced the API to take a material elevation instead of the raw DrawShadow parameters.  PTAL.

**Comment by abarth** [comment]
LGTM

**Pull request merged**